docs/expression_syntax.md

```diff
@@ -0,0 +1,195 @@
+**Is your feature request related to a problem? Please describe.**
```

Comment:
I think we should just describe the currently supported expression syntax. We can direct users who want to know about upcoming changes to the RFC at the bottom of this document.